### PR TITLE
Migrate Java 8 to 11 and  upgrade SprintBoot Framework version

### PR DESCRIPTION
**Pull request by dan-du-car**
# PR Details
## Description
<meta name="ProgId" content="Word.Document">
<meta name="Generator" content="Microsoft Word 15">
<meta name="Originator" content="Microsoft Word 15">
<link rel="File-List" href="file:///C:/Users/dud/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<link rel="themeData" href="file:///C:/Users/dud/AppData/Local/Temp/msohtmlclip1/01/clip_themedata.thmx">
<link rel="colorSchemeMapping" href="file:///C:/Users/dud/AppData/Local/Temp/msohtmlclip1/01/clip_colorschememapping.xml">

<style>

</style>

<p class="MsoListParagraph" style="margin-bottom:0in;margin-bottom:.0001pt;
mso-add-space:auto;text-align:justify;text-justify:inter-ideograph;text-indent:
-.25in;line-height:normal;mso-list:l0 level1 lfo1"><span style="font-size:14.0pt;font-family:Symbol;mso-fareast-font-family:Symbol;
mso-bidi-font-family:Symbol"><span style="mso-list:Ignore">·<span style="font:7.0pt &quot;Times New Roman&quot;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><span style="font-size:14.0pt;font-family:&quot;Times New Roman&quot;,serif">Current</span><span style="font-size:14.0pt;font-family:&quot;Times New Roman&quot;,serif;mso-fareast-font-family:
&quot;Times New Roman&quot;;color:#292929;letter-spacing:-.05pt"> versions of Sprint Boot,
Sprint, Maven compiler, etc. The upcoming migration versions list below: </span><span style="font-size:14.0pt;font-family:&quot;Times New Roman&quot;,serif"><o:p></o:p></span></p>

&nbsp; | Current Version | Migration to Version
-- | -- | --
JDK | 8 | 11
Spring Framework | 4.3 | 5.3.6
Spring Boot | 1.4.1.RELEASE | 2.4.5
Maven Compiler | 3.1 | 3.8.1
JUnit | 4.12 | 4.13.2
JAXB | &nbsp; | 2.3.1
Maven Release Plugin | 2.5.3 | 3.0.0-M4

## Related Issue
N/A

## Motivation and Context
Java version upgrade to 11

## How Has This Been Tested?
Local integration testing

## Types of changes
Java library version and sprint boot framework upgrade

- [ ] Defect fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
[ODE Contributing Guide](https://github.com/usdot-jpo-ode/jpo-ode/blob/bugfix/Pull_request_template/docs/contributing_guide.md) 
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by drewjj — DISMISSED** [review]
Tested for CDOT use cases and everything seems to work fine.

**Comment by drewjj** [comment]
Apologies on the testing I performed previously. CDOT runs the ODE in GCP Kubernetes and Kubernetes was not behaving as I was expecting. I ended up testing on an older working version of the ODE which is no good. Now that this has been resolved, I have properly tested the new J11 ODE for CDOT's test cases.

Test cases:
- Send BSM over UDP port 46800 to ODE
- Send TIM over UDP port 47900 to ODE

Expected:
- BSM decoded and published to Kafka topic topic.OdeBsmJson
- TIM decoded and published to Kafka topic topic.OdeTimJson

Results:
Both tests were successful.

**BSM Test decoded output from topic.OdeBsmJson**
{"metadata":{"originIp":"172.16.28.221","bsmSource":"RV","logFileName":"","recordType":"bsmTx","securityResultCode":"success","receivedMessageDetails":{"locationData":{"latitude":"","longitude":"","elevation":"","speed":"","heading":""},"rxSource":"RV"},"payloadType":"us.dot.its.jpo.ode.model.OdeBsmPayload","serialId":{"streamId":"2403feec-3c19-40fb-b774-051ce035d808","bundleSize":1,"bundleId":0,"recordId":0,"serialNumber":0},"odeReceivedAt":"2021-06-07T18:13:22.415149Z","schemaVersion":6,"maxDurationTime":0,"odePacketID":"","odeTimStartDateTime":"","recordGeneratedAt":"","sanitized":false},"payload":{"dataType":"us.dot.its.jpo.ode.plugin.j2735.J2735Bsm","data":{"coreData":{"msgCnt":122,"id":"3626240C","secMark":33500,"position":{"latitude":39.7649302,"longitude":-105.6314486,"elevation":2478.7},"accelSet":{"accelLong":1.70,"accelYaw":0.00},"accuracy":{"semiMajor":2.00,"semiMinor":2.00,"orientation":44.4953079900},"transmission":"NEUTRAL","speed":29.96,"heading":61.5000,"brakes":{"wheelBrakes":{"leftFront":false,"rightFront":false,"unavailable":false,"leftRear":false,"rightRear":false},"traction":"unavailable","abs":"unavailable","scs":"unavailable","brakeBoost":"unavailable","auxBrakes":"unavailable"},"size":{"width":260,"length":1020}},"partII":[{"id":"VehicleSafetyExtensions","value":{"pathHistory":{"crumbData":[{"elevationOffset":-0.3,"latOffset":0.0002959,"lonOffset":0.0007440,"timeOffset":2.40},{"elevationOffset":0.0,"latOffset":0.0005657,"lonOffset":0.0015725,"timeOffset":5.00},{"elevationOffset":0.4,"latOffset":0.0007741,"lonOffset":0.0024216,"timeOffset":7.60},{"elevationOffset":1.7,"latOffset":0.0012313,"lonOffset":0.0047935,"timeOffset":14.80}]},"pathPrediction":{"confidence":90.0,"radiusOfCurve":-1931.4},"lights":{"daytimeRunningLightsOn":false,"automaticLightControlOn":true,"rightTurnSignalOn":false,"parkingLightsOn":false,"lowBeamHeadlightsOn":false,"hazardSignalOn":false,"highBeamHeadlightsOn":false,"leftTurnSignalOn":false,"fogLightOn":false}}},{"id":"SpecialVehicleExtensions","value":{"vehicleAlerts":{"sspRights":0,"lightsUse":"NOTINUSE","multi":"UNAVAILABLE","sirenUse":"NOTINUSE"}}}]}}}

**TIM Test decoded output from topic.OdeTimJson**
{"metadata":{"securityResultCode":"","recordGeneratedBy":"RSU","schemaVersion":6,"odePacketID":"","sanitized":false,"recordGeneratedAt":"","recordType":"timMsg","maxDurationTime":0,"odeTimStartDateTime":"","receivedMessageDetails":"","payloadType":"us.dot.its.jpo.ode.model.OdeTimPayload","serialId":{"recordId":0,"serialNumber":0,"streamId":"92bca0eb-1c59-4856-bf40-1d158b10c288","bundleSize":1,"bundleId":0},"logFileName":"","odeReceivedAt":"2021-06-07T18:18:06.938495Z","originIp":"172.16.28.221"},"payload":{"data":{"MessageFrame":{"messageId":31,"value":{"TravelerInformation":{"packetID":"00000000000564FB69","dataFrames":{"TravelerDataFrame":{"regions":{"GeographicalPath":{"closedPath":{"false":""},"anchor":{"lat":404741206,"long":-1049694889},"laneWidth":1000,"directionality":{"both":""},"description":{"path":{"offset":{"xy":{"nodes":{"NodeXY":[{"delta":{"node-LatLon":{"lon":-1049683204,"lat":404748120}}},{"delta":{"node-LatLon":{"lon":-1049689644,"lat":404744920}}},{"delta":{"node-LatLon":{"lon":-1049702575,"lat":404735755}}},{"delta":{"node-LatLon":{"lon":-1049707300,"lat":404734096}}}]}}}}},"direction":1111111111111111}},"duratonTime":10080,"sspMsgRights1":1,"sspMsgRights2":1,"sspTimRights":1,"startYear":2021,"sspLocationRights":1,"frameType":{"advisory":""},"msgId":{"roadSignID":{"viewAngle":1111111111111111,"mutcdCode":{"warning":""},"position":{"lat":404741206,"long":-1049694889}}},"startTime":57221,"priority":5,"content":{"advisory":{"SEQUENCE":{"item":{"itis":5906}}}}}},"msgCnt":1}}}},"dataType":"TravelerInformation"}}

**Comment by dan-du-car** [comment]
> Apologies on the testing I performed previously. CDOT runs the ODE in GCP Kubernetes and Kubernetes was not behaving as I was expecting. I ended up testing on an older working version of the ODE which is no good. Now that this has been resolved, I have properly tested the new J11 ODE for CDOT's test cases.
> 
> Test cases:
> 
> * Send BSM over UDP port 46800 to ODE
> * Send TIM over UDP port 47900 to ODE
> 
> Expected:
> 
> * BSM decoded and published to Kafka topic topic.OdeBsmJson
> * TIM decoded and published to Kafka topic topic.OdeTimJson
> 
> Results:
> Both tests were successful.
> 
> **BSM Test decoded output from topic.OdeBsmJson**
> {"metadata":{"originIp":"172.16.28.221","bsmSource":"RV","logFileName":"","recordType":"bsmTx","securityResultCode":"success","receivedMessageDetails":{"locationData":{"latitude":"","longitude":"","elevation":"","speed":"","heading":""},"rxSource":"RV"},"payloadType":"us.dot.its.jpo.ode.model.OdeBsmPayload","serialId":{"streamId":"2403feec-3c19-40fb-b774-051ce035d808","bundleSize":1,"bundleId":0,"recordId":0,"serialNumber":0},"odeReceivedAt":"2021-06-07T18:13:22.415149Z","schemaVersion":6,"maxDurationTime":0,"odePacketID":"","odeTimStartDateTime":"","recordGeneratedAt":"","sanitized":false},"payload":{"dataType":"us.dot.its.jpo.ode.plugin.j2735.J2735Bsm","data":{"coreData":{"msgCnt":122,"id":"3626240C","secMark":33500,"position":{"latitude":39.7649302,"longitude":-105.6314486,"elevation":2478.7},"accelSet":{"accelLong":1.70,"accelYaw":0.00},"accuracy":{"semiMajor":2.00,"semiMinor":2.00,"orientation":44.4953079900},"transmission":"NEUTRAL","speed":29.96,"heading":61.5000,"brakes":{"wheelBrakes":{"leftFront":false,"rightFront":false,"unavailable":false,"leftRear":false,"rightRear":false},"traction":"unavailable","abs":"unavailable","scs":"unavailable","brakeBoost":"unavailable","auxBrakes":"unavailable"},"size":{"width":260,"length":1020}},"partII":[{"id":"VehicleSafetyExtensions","value":{"pathHistory":{"crumbData":[{"elevationOffset":-0.3,"latOffset":0.0002959,"lonOffset":0.0007440,"timeOffset":2.40},{"elevationOffset":0.0,"latOffset":0.0005657,"lonOffset":0.0015725,"timeOffset":5.00},{"elevationOffset":0.4,"latOffset":0.0007741,"lonOffset":0.0024216,"timeOffset":7.60},{"elevationOffset":1.7,"latOffset":0.0012313,"lonOffset":0.0047935,"timeOffset":14.80}]},"pathPrediction":{"confidence":90.0,"radiusOfCurve":-1931.4},"lights":{"daytimeRunningLightsOn":false,"automaticLightControlOn":true,"rightTurnSignalOn":false,"parkingLightsOn":false,"lowBeamHeadlightsOn":false,"hazardSignalOn":false,"highBeamHeadlightsOn":false,"leftTurnSignalOn":false,"fogLightOn":false}}},{"id":"SpecialVehicleExtensions","value":{"vehicleAlerts":{"sspRights":0,"lightsUse":"NOTINUSE","multi":"UNAVAILABLE","sirenUse":"NOTINUSE"}}}]}}}
> 
> **TIM Test decoded output from topic.OdeTimJson**
> {"metadata":{"securityResultCode":"","recordGeneratedBy":"RSU","schemaVersion":6,"odePacketID":"","sanitized":false,"recordGeneratedAt":"","recordType":"timMsg","maxDurationTime":0,"odeTimStartDateTime":"","receivedMessageDetails":"","payloadType":"us.dot.its.jpo.ode.model.OdeTimPayload","serialId":{"recordId":0,"serialNumber":0,"streamId":"92bca0eb-1c59-4856-bf40-1d158b10c288","bundleSize":1,"bundleId":0},"logFileName":"","odeReceivedAt":"2021-06-07T18:18:06.938495Z","originIp":"172.16.28.221"},"payload":{"data":{"MessageFrame":{"messageId":31,"value":{"TravelerInformation":{"packetID":"00000000000564FB69","dataFrames":{"TravelerDataFrame":{"regions":{"GeographicalPath":{"closedPath":{"false":""},"anchor":{"lat":404741206,"long":-1049694889},"laneWidth":1000,"directionality":{"both":""},"description":{"path":{"offset":{"xy":{"nodes":{"NodeXY":[{"delta":{"node-LatLon":{"lon":-1049683204,"lat":404748120}}},{"delta":{"node-LatLon":{"lon":-1049689644,"lat":404744920}}},{"delta":{"node-LatLon":{"lon":-1049702575,"lat":404735755}}},{"delta":{"node-LatLon":{"lon":-1049707300,"lat":404734096}}}]}}}}},"direction":1111111111111111}},"duratonTime":10080,"sspMsgRights1":1,"sspMsgRights2":1,"sspTimRights":1,"startYear":2021,"sspLocationRights":1,"frameType":{"advisory":""},"msgId":{"roadSignID":{"viewAngle":1111111111111111,"mutcdCode":{"warning":""},"position":{"lat":404741206,"long":-1049694889}}},"startTime":57221,"priority":5,"content":{"advisory":{"SEQUENCE":{"item":{"itis":5906}}}}}},"msgCnt":1}}}},"dataType":"TravelerInformation"}}

Awesome!

File: jpo-ode-plugins/src/test/java/us/dot/its/jpo/ode/plugin/j2735/builders/TravelerMessageFromHumanToAsnConverterTest.java [hunk 1]
```diff
@@ -63,8 +64,21 @@ public void testAdvisoryNodeLL() throws JsonProcessingException, IOException, Js
...
     assertEquals(expectedTID.toString(), inputTID.toString());
     JSONObject timObject = new JSONObject();
-    timObject.put(TravelerMessageFromHumanToAsnConverter.TRAVELER_INFORMATION, JsonUtils.toJSONObject(inputTID.toString()));
-    assertNotNull(XML.toString(timObject));
+    try {
+		timObject.put(TravelerMessageFromHumanToAsnConverter.TRAVELER_INFORMATION, JsonUtils.toJSONObject(inputTID.toString()));
+	} catch (JSONException e1) {
+		// TODO Auto-generated catch block
```
Comment (snallamothu) [inline jpo-ode-plugins/src/test/java/us/dot/its/jpo/ode/plugin/j2735/builders/TravelerMessageFromHumanToAsnConverterTest.java:70]: Is there more todo in the catch blocks other than printing the stack trace.  If not, delete the TODO statements next time to keep the code clean here and in other files where applicable.

Comment (dan-du-car) [reply jpo-ode-plugins/src/test/java/us/dot/its/jpo/ode/plugin/j2735/builders/TravelerMessageFromHumanToAsnConverterTest.java]: addressed 54c7dd2

**Comment by payneBrandon** [comment]
I've ran into a few issues with testing in both CDOT and WYDOT environments. When attempting to send previously valid JSON (as verified in the `release/1.2.1` tag) through the `/tim` endpoint I'm receiving the attached error. This seems to boil down to the `timstorage/Position.java` class no longer understanding the "latitude" property.
[error.log](https://github.com/usdot-jpo-ode/jpo-ode/files/6618369/error.log)

**Comment by payneBrandon** [comment]
> I've ran into a few issues with testing in both CDOT and WYDOT environments. When attempting to send previously valid JSON (as verified in the `release/1.2.1` tag) through the `/tim` endpoint I'm receiving the attached error. This seems to boil down to the `timstorage/Position.java` class no longer understanding the "latitude" property.
> [error.log](https://github.com/usdot-jpo-ode/jpo-ode/files/6618369/error.log)
> @payneBrandon Can you attach the JSON data you send to the /tim endpoint?
> 
> @dan-du-car The JSON data is in the first part of the log file (line 7 I believe, the ODE logs this on receipt)- I made sure to include it in there for you to test.

@dan-du-car To elaborate on this a bit, below is a step by step I used to run this same test:

- checkout `feature/upgrade-java-11` branch
```
git checkout feature/upgrade-java-11
git pull --recurse-submodules
git submodule update --recursive --init
```

- run script to restart ODE
```
./scripts/clean-build-and-deploy.sh
```

- once ODE is back up and running send the TIM from the attached log file to the ODE `/tim` endpoint. CURL example:
```
curl --location --request POST 'my-running-ode-implementation:8080/tim' \
--header 'Content-Type: application/json' \
--data '{
    "request": {
        "ode": {
            "verb": "POST",
            "version": 3
        },
        "sdw": {
            "recordId": "773D90DB",
            "serviceRegion": {
                "nwCorner": {
                    "latitude": 41.35935118,
                    "longitude": -105.76248582
                },
                "seCorner": {
                    "latitude": 41.33446431,
                    "longitude": -105.61656942
                }
            },
            "ttl": "oneyear"
        }
    },
    "tim": {
        "dataFrames": {
            "TravelerDataFrame": [
                {
                    "duratonTime": 32000,
                    "frameType": {
                        "advisory": "EMPTY_TAG"
                    },
                    "msgId": {
                        "roadSignID": {
                            "mutcdCode": {
                                "warning": "EMPTY_TAG"
                            },
                            "position": {
                                "elevation": null,
                                "latitude": 413598379,
                                "longitude": -1057643057
                            },
                            "viewAngle": "1111111111111111"
                        }
                    },
                    "priority": 5,
                    "regions": {
                        "GeographicalPath": [
                            {
                                "anchor": {
                                    "lat": 413598379,
                                    "long": -1057643057
                                },
                                "closedPath": "BOOLEAN_OBJECT_FALSE",
                                "description": {
                                    "path": {
                                        "offset": {
                                            "ll": {
                                                "nodes": {
                                                    "NodeLL": [
                                                        {
                                                            "delta": {
                                                                "node-LL3": {
                                                                    "lat": -4868,
                                                                    "lon": 18199
                                                                }
                                                            }
                                                        },
                                                        {
                                                            "delta": {
                                                                "node-LL5": {
                                                                    "lat": -47533,
                                                                    "lon": 182455
                                                                }
                                                            }
                                                        },
                                                        {
                                                            "delta": {
                                                                "node-LL5": {
                                                                    "lat": -55912,
                                                                    "lon": 395806
                                                                }
                                                            }
                                                        },
                                                        {
                                                            "delta": {
                                                                "node-LL4": {
                                                                    "lat": 3894,
                                                                    "lon": 37790
                                                                }
                                                            }
                                                        },
                                                        {
                                                            "delta": {
                                                                "node-LL4": {
                                                                    "lat": 10579,
                                                                    "lon": 55552
                                                                }
                                                            }
                                                        },
                                                        {
                                                            "delta": {
                                                                "node-LL5": {
                                                                    "lat": 259,
                                                                    "lon": 517562
                                                                }
                                                            }
                                                        },
                                                        {
                                                            "delta": {
                                                                "node-LL4": {
                                                                    "lat": -7404,
                                                                    "lon": 36939
                                                                }
                                                            }
                                                        },
                                                        {
                                                            "delta": {
                                                                "node-LL5": {
                                                                    "lat": -88890,
                                                                    "lon": 197431
                                                                }
                                                            }
                                                        },
                                                        {
                                                            "delta": {
                                                                "node-LL3": {
                                                                    "lat": -21647,
                                                                    "lon": 24872
                                                                }
                                                            }
                                                        },
                                                        {
                                                            "delta": {
                                                                "node-LL3": {
                                                                    "lat": -27795,
                                                                    "lon": 9458
                                                                }
                                                            }
                                                        },
                                                        {
                                                            "delta": {
                                                                "node-LL3": {
                                                                    "lat": -14420,
                                                                    "lon": 1301
                                                                }
                                                            }
                                                        }
                                                    ]
                                                }
                                            }
                                        },
                                        "scale": 0
                                    }
                                },
                                "direction": "0001111100000000",
                                "directionality": {
                                    "both": "EMPTY_TAG"
                                },
                                "id": {
                                    "id": 0,
                                    "region": 0
                                },
                                "laneWidth": 5000,
                                "name": "I_I 80_SAT-773D90DB_RC_STRONGWINDS-TEST-Brandon-0"
                            }
                        ]
                    },
                    "sspLocationRights": 1,
                    "sspMsgRights1": 1,
                    "sspMsgRights2": 1,
                    "sspTimRights": "1",
                    "startTime": 228459,
                    "startYear": 2021,
                    "tcontent": {
                        "advisory": {
                            "SEQUENCE": [
                                {
                                    "item": {
                                        "itis": 5127
                                    }
                                }
                            ]
                        }
                    },
                    "url": "null"
                }
            ]
        },
        "msgCnt": 1,
        "packetID": "C1A37ECD94C46BEB2E",
        "timeStamp": 228459,
        "urlB": "null"
    }
}'
```

This produces the attached error log from the ODE.

**Comment by dan-du-car** [comment]
> > I've ran into a few issues with testing in both CDOT and WYDOT environments. When attempting to send previously valid JSON (as verified in the `release/1.2.1` tag) through the `/tim` endpoint I'm receiving the attached error. This seems to boil down to the `timstorage/Position.java` class no longer understanding the "latitude" property.
> > [error.log](https://github.com/usdot-jpo-ode/jpo-ode/files/6618369/error.log)
> > @payneBrandon Can you attach the JSON data you send to the /tim endpoint?
> > @dan-du-car The JSON data is in the first part of the log file (line 7 I believe, the ODE logs this on receipt)- I made sure to include it in there for you to test.
> 
> @dan-du-car To elaborate on this a bit, below is a step by step I used to run this same test:
> 
> * checkout `feature/upgrade-java-11` branch
> 
> ```
> git checkout feature/upgrade-java-11
> git pull --recurse-submodules
> git submodule update --recursive --init
> ```
> 
> * run script to restart ODE
> 
> ```
> ./scripts/clean-build-and-deploy.sh
> ```
> 
> * once ODE is back up and running send the TIM from the attached log file to the ODE `/tim` endpoint. CURL example:
> 
> ```
> curl --location --request POST 'my-running-ode-implementation:8080/tim' \
> --header 'Content-Type: application/json' \
> --data '{
>     "request": {
>         "ode": {
>             "verb": "POST",
>             "version": 3
>         },
>         "sdw": {
>             "recordId": "773D90DB",
>             "serviceRegion": {
>                 "nwCorner": {
>                     "latitude": 41.35935118,
>                     "longitude": -105.76248582
>                 },
>                 "seCorner": {
>                     "latitude": 41.33446431,
>                     "longitude": -105.61656942
>                 }
>             },
>             "ttl": "oneyear"
>         }
>     },
>     "tim": {
>         "dataFrames": {
>             "TravelerDataFrame": [
>                 {
>                     "duratonTime": 32000,
>                     "frameType": {
>                         "advisory": "EMPTY_TAG"
>                     },
>                     "msgId": {
>                         "roadSignID": {
>                             "mutcdCode": {
>                                 "warning": "EMPTY_TAG"
>                             },
>                             "position": {
>                                 "elevation": null,
>                                 "latitude": 413598379,
>                                 "longitude": -1057643057
>                             },
>                             "viewAngle": "1111111111111111"
>                         }
>                     },
>                     "priority": 5,
>                     "regions": {
>                         "GeographicalPath": [
>                             {
>                                 "anchor": {
>                                     "lat": 413598379,
>                                     "long": -1057643057
>                                 },
>                                 "closedPath": "BOOLEAN_OBJECT_FALSE",
>                                 "description": {
>                                     "path": {
>                                         "offset": {
>                                             "ll": {
>                                                 "nodes": {
>                                                     "NodeLL": [
>                                                         {
>                                                             "delta": {
>                                                                 "node-LL3": {
>                                                                     "lat": -4868,
>                                                                     "lon": 18199
>                                                                 }
>                                                             }
>                                                         },
>                                                         {
>                                                             "delta": {
>                                                                 "node-LL5": {
>                                                                     "lat": -47533,
>                                                                     "lon": 182455
>                                                                 }
>                                                             }
>                                                         },
>                                                         {
>                                                             "delta": {
>                                                                 "node-LL5": {
>                                                                     "lat": -55912,
>                                                                     "lon": 395806
>                                                                 }
>                                                             }
>                                                         },
>                                                         {
>                                                             "delta": {
>                                                                 "node-LL4": {
>                                                                     "lat": 3894,
>                                                                     "lon": 37790
>                                                                 }
>                                                             }
>                                                         },
>                                                         {
>                                                             "delta": {
>                                                                 "node-LL4": {
>                                                                     "lat": 10579,
>                                                                     "lon": 55552
>                                                                 }
>                                                             }
>                                                         },
>                                                         {
>                                                             "delta": {
>                                                                 "node-LL5": {
>                                                                     "lat": 259,
>                                                                     "lon": 517562
>                                                                 }
>                                                             }
>                                                         },
>                                                         {
>                                                             "delta": {
>                                                                 "node-LL4": {
>                                                                     "lat": -7404,
>                                                                     "lon": 36939
>                                                                 }
>                                                             }
>                                                         },
>                                                         {
>                                                             "delta": {
>                                                                 "node-LL5": {
>                                                                     "lat": -88890,
>                                                                     "lon": 197431
>                                                                 }
>                                                             }
>                                                         },
>                                                         {
>                                                             "delta": {
>                                                                 "node-LL3": {
>                                                                     "lat": -21647,
>                                                                     "lon": 24872
>                                                                 }
>                                                             }
>                                                         },
>                                                         {
>                                                             "delta": {
>                                                                 "node-LL3": {
>                                                                     "lat": -27795,
>                                                                     "lon": 9458
>                                                                 }
>                                                             }
>                                                         },
>                                                         {
>                                                             "delta": {
>                                                                 "node-LL3": {
>                                                                     "lat": -14420,
>                                                                     "lon": 1301
>                                                                 }
>                                                             }
>                                                         }
>                                                     ]
>                                                 }
>                                             }
>                                         },
>                                         "scale": 0
>                                     }
>                                 },
>                                 "direction": "0001111100000000",
>                                 "directionality": {
>                                     "both": "EMPTY_TAG"
>                                 },
>                                 "id": {
>                                     "id": 0,
>                                     "region": 0
>                                 },
>                                 "laneWidth": 5000,
>                                 "name": "I_I 80_SAT-773D90DB_RC_STRONGWINDS-TEST-Brandon-0"
>                             }
>                         ]
>                     },
>                     "sspLocationRights": 1,
>                     "sspMsgRights1": 1,
>                     "sspMsgRights2": 1,
>                     "sspTimRights": "1",
>                     "startTime": 228459,
>                     "startYear": 2021,
>                     "tcontent": {
>                         "advisory": {
>                             "SEQUENCE": [
>                                 {
>                                     "item": {
>                                         "itis": 5127
>                                     }
>                                 }
>                             ]
>                         }
>                     },
>                     "url": "null"
>                 }
>             ]
>         },
>         "msgCnt": 1,
>         "packetID": "C1A37ECD94C46BEB2E",
>         "timeStamp": 228459,
>         "urlB": "null"
>     }
> }'
> ```
> 
> This produces the attached error log from the ODE.

This is addressed by https://github.com/usdot-jpo-ode/jpo-ode/pull/441/commits/ba75113efb01ecde07d0ac1cbc1411ffbb3a7360

**Review by payneBrandon — APPROVED** [review]
This looks good to me. I've successfully tested this branch in both WYDOT and CDOT development environments sending various TIM and BSM messages.

**Comment by drewjj** [comment]
Looks good to me as well. Tested this with CDOT use cases in CDOT dev environment and it behaves as it should.

Great work!

**Review by drewjj — APPROVED** [review]

**Review by TonyEnglish — APPROVED** [review]

**Pull request merged**